src/Common/src/TypeSystem/Interop/IL/Marshaller.cs

```diff
@@ -1681,9 +1747,17 @@ protected override void TransformManagedToNative(ILCodeStream codeStream)
         }
         protected override void AllocNativeToManaged(ILCodeStream codeStream)
         {
-            codeStream.Emit(ILOpcode.call, _ilCodeStreams.Emitter.NewToken(
+            var emitter = _ilCodeStreams.Emitter;
```

Comment:
Might be intersting to abstract this into a base class (such as ReferenceMarshaler)